### PR TITLE
Train general component model on a single Fenix class

### LINKED ISSUE #4229: Train general component model on a single Fenix class to improve accuracy
The current [general component model](https://github.com/mozilla/bugbug/blob/master/bugbug/models/component.py) trained on all Fenix components. Due to the low number of bugs in some components and similarity with components in other products, the model's accuracy is limited. A proposal is to group all Fenix components into a single Fenix class for training the general model, which can later be refined with the [Fenix-specific component model](https://github.com/mozilla/bugbug/blob/master/bugbug/models/fenixcomponent.py).

### PR DESCRIPTION
**Pull request by benjaminmah**
Resolves #4229.

Groups all Fenix components into a single Fenix class (i.e. label all Fenix bugs as `Fenix`).

### CODE REVIEW

**Comment by benjaminmah** [comment]
[Training Metrics](https://github.com/user-attachments/files/15777117/output.log)

File: bugbug/models/component.py [hunk 1]
```diff
@@ -132,6 +132,8 @@ def __init__(self, lemmatization=False):
 
     def filter_component(self, product, component):
         full_comp = f"{product}::{component}"
+        if product == "Fenix":
+            return "Fenix"
```
Comment (suhaibmujahid) [inline bugbug/models/component.py:136]: I would move this to the beginning of the method (i.e., before `full_comp`).

**Review by suhaibmujahid — COMMENTED** [review]
What was the aggregated percussion and recall for Fenix components  before this PR?

File: bugbug/models/component.py (hunk 1 above)
Comment (benjaminmah) [reply bugbug/models/component.py:135]: Done in 896d05dd6a91069c918c2a51f55d9292c6669344!

Comment (benjaminmah) [reply bugbug/models/component.py:135]: Will generate metrics on the original model before this PR ASAP.

**Comment by benjaminmah** [comment]
Aggregated results of the model before and after:
[Component Model Metrics](https://docs.google.com/document/d/11tk2N7A0GxIvA8aWorqt6eSuzKgZQpPG6_8yycH8wmM/edit?usp=sharing)

**Review by suhaibmujahid — APPROVED** [review]
The results looks better. Thank you!

**Pull request merged**